rbac/management/role/view.py

```diff
@@ -639,6 +641,20 @@ def validate_role(self, request):
                         error = {key: [_(message)]}
                         raise serializers.ValidationError(error)
 
+                # validate permission not being added to workspace out of users org for v1 (RHCLOUD-35481)
+                resourceDefinitions = perm.get("resourceDefinitions")
+                for resourceDefinition in resourceDefinitions:
+                    attributeFilter = resourceDefinition.get("attributeFilter")
+                    if is_resource_a_workspace(app, resource_type, attributeFilter):
+                        workspace_id = get_workspace_id_from_resource_definition(attributeFilter)
+                        is_same_tenant = Workspace.objects.filter(id=workspace_id, tenant=request.tenant).exists()
```

Comment:
Yeah, I wonder if it's worth just having this always return a list, and doing a check like:
```suggestion
                        workspace_ids = get_workspace_ids_from_resource_definition(attributeFilter)
                        is_same_tenant = Workspace.objects.filter(id__in=workspace_ids, tenant=request.tenant).exists()
```

Where then `get_workspace_ids_from_resource_definition` then always returns a list regardless of what's passed in?

Reply:
thats the change I made :)